File: static/textSanitizer.js
// This will sanitize text of foreign JavaScript code and SQL commands


function sanitizeData(text) {
    var cleanText = text;
    // General syntax breaker
    cleanText = cleanText.replaceAll(";", "");
    // Replace HTML characters
    cleanText = cleanText.replaceAll("&", "&amp");
    cleanText = cleanText.replaceAll("<", "&lt");
    cleanText = cleanText.replaceAll(">", "&gt");
    do {
        // Finish off with SQL commands
        cleanText = cleanText.replaceAll("SELECT", "");
        cleanText = cleanText.replaceAll("select", "");
        cleanText = cleanText.replaceAll("CREATE", "");
        cleanText = cleanText.replaceAll("create", "");
        cleanText = cleanText.replaceAll("UPDATE", "");
        cleanText = cleanText.replaceAll("update", "");
        cleanText = cleanText.replaceAll("DELETE", "");
        cleanText = cleanText.replaceAll("delete", "");
    } while (cleanText.toLowerCase().includes("select") && cleanText.toLowerCase().includes("create") &&
    	cleanText.toLowerCase().includes("update") && cleanText.toLowerCase().includes("delete"));
    // Stripping of additional spaces
    for (i=cleanText.length; i>0; i--) {
	if (cleanText[i]==' '&&cleanText[i-1]==' ') {
	    cleanText = cleanText.substring(0,i-1) + cleanText.substring(i+1);
	}
    }
    alert("Search string: " + cleanText); // For testing purposes
    return cleanText;
}

File: textSanitizer.py
# This will sanitize text of foreign JavaScript code and SQL commands


def sanitizeData(text):
    cleanText = text
    # General syntax breaker
    cleanText = cleanText.replace(";", "")
    # Replace HTML characters
    cleanText = cleanText.replace("&", "&amp")
    cleanText = cleanText.replace("<", "&lt")
    cleanText = cleanText.replace(">", "&gt")
    while (cleanText.lower().find("select")!=-1 or cleanText.lower().find("create")!=-1 or
        cleanText.lower().find("update")!=-1 or cleanText.lower().find("delete")!=-1):
        # Finish off with SQL commands
        cleanText = cleanText.replace("SELECT", "")
        cleanText = cleanText.replace("select", "")
        cleanText = cleanText.replace("CREATE", "")
        cleanText = cleanText.replace("create", "")
        cleanText = cleanText.replace("UPDATE", "")
        cleanText = cleanText.replace("update", "")
        cleanText = cleanText.replace("DELETE", "")
        cleanText = cleanText.replace("delete", "")
    # Stripping of additional spaces
    for i in range(len(cleanText)-1, 0, -1):
        if cleanText[i]==" " and cleanText[i-1]==" ":
            cleanText = cleanText[:i-1] + cleanText[i:]
    return cleanText
